networkx/generators/stochastic.py

```diff
@@ -50,4 +50,5 @@ def stochastic_graph(G, copy=True, weight="weight"):
             d[weight] = 0
         else:
             d[weight] = d.get(weight, 1) / degree[u]
+    G.__networkx_cache__.clear()
```

Comment:
For the `stochastic_graph`, I think the fix is to change from `copy` making an entirely new graph, to allowing it to create a new attribute that holds the stochastic weight. There is no reason to copy the whole graph just  to save the original weight values.